src/components/todoItem.js

```diff
@@ -1,22 +1,31 @@
 import React from 'react';
 import {observer} from 'mobx-react';
+import {observable, expr} from 'mobx';
 
 const ESCAPE_KEY = 27;
 const ENTER_KEY = 13;
 
+
+class State {
+	@observable editText = '';
+	constructor(editText){
+		this.editText = editText;
+	}
+}
+
 @observer
 export default class TodoItem extends React.Component {
 	constructor(props, context) {
 		super(props, context);
-		this.state = {editText: props.todo.title};
+		this.state = new State(props.todo.title);
```

Comment:
That's the question of who maintains the component and under which assumptions. Keeping editText outside of the component's state can confuse the unwary debugger / dev tools.